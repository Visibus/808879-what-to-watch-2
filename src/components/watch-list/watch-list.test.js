import WatchList from "./watch-list";
import {BrowserRouter} from "react-router-dom";
import films from "../../mocks/films.js";

it(`FavoriteList component renders correctly`, () => {
  const tree = window.renderer
    .create(<BrowserRouter>
      <WatchList
        isAuthorizationRequired={true}
        userData={{id: 1, name: `fake`, email: `test@mail.ru`, avatarUrl: ``}}
        films={films}
      >
      </WatchList>
    </BrowserRouter>)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
