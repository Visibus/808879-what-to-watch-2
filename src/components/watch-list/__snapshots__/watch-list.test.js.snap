// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList component renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/1"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/2"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/3"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/4"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/5"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/6"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/6"
            onClick={[Function]}
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/7"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/7"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/8"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/8"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/9"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/9"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/10"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/10"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/11"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/11"
            onClick={[Function]}
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/12"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/12"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/13"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/13"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/14"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/14"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/15"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/15"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/16"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/16"
            onClick={[Function]}
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/17"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/17"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/18"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/18"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/19"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/19"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/20"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/20"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/21"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/seven-years-in-tibet.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/21"
            onClick={[Function]}
          >
            Seven Years in Tibet
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/22"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/22"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/23"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/23"
            onClick={[Function]}
          >
            Johnny English
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/24"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/24"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          href="/films/25"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width={280}
            />
          </div>
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/25"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
  </section>
</div>
`;
